### DISCUSSION TITLE
Migrations with sequelize and enums

### DISCUSSION THREAD

**Discussion by lededje** [sequelize/umzug#547] [General]
An issue that appears when using umuz and sequelize when you have enums in your migrations and models.

```
"dependencies": {
    "pg": "^8.7.3",
    "pg-hstore": "^2.3.4",
    "sequelize": "^6.17.0",
    "umzug": "^3.0.0"
  }
```

Minimum viable example: https://github.com/lededje/migration-bug

```
docker-compose up
node index.js
```

expected result: Migrations are run up and then down

actual result: 

```
➜  test git:(main) node index                     
{ event: 'reverting', name: '0001-create-tags' }  
                                                  
/home/lededje/repos/test/node_modules/umzug/lib/umzug.js:205
                    throw new MigrationError({ direction: 'down', ...params }, e);
                          ^                       
MigrationError: Migration 0001-create-tags (down) failed: Original error: Cannot set properties of un
defined (setting 'supportsSearchPath')       
    at /home/lededje/repos/test/node_modules/umzug/lib/umzug.js:205:27                               
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Umzug.runCommand (/home/lededje/repos/test/node_modules/umzug/lib/umzug.js:110:20)
    at async /home/lededje/repos/test/index.js:76:3 {
  jse_shortmsg: 'Migration 0001-create-tags (down) failed',
  jse_cause: VError: Original error: Cannot set properties of undefined (setting 'supportsSearchPath'
)                                                 
```
If you checkout the `working` branch, and repeat the test (with a clean db) the migrations run as expected.

**Comment by lededje** [comment]
After further investigation, it appears to also be fixed by providing umzug a separate sequelize instance to the one you use for models. This issue occurred when I was unit testing, so the unit tests and umzug shared a sequelize instance which was causing the problem.

The impact of this fix is small so I'm happy for this issue to be closed but I am curious why this happens.

**Comment by mmkal** [comment]
A note in the docs about this would be very welcome. Umzug isn’t coupled to sequelize so if you use sequelize you need to be aware about how it caches models etc.

**Comment by tanmayaBiswalOdiware** [comment]
```js
MigrationError: Migration 00000_initial.ts (down) failed: Original error: Cannot set properties of undefined (setting 'supportsSearchPath')

```

I am completely clueless as to why I am getting this error. Any ideas from you guys? Or how I could get around it?

**Comment by Adiotastic** [comment]
I had this issue for dropTable in my up() function.
After further investigation, I saw, that if you we use `....dropTable('TableName')` without the options object as a second parameter, and we have an enum in our model, it fails. Umzug tries to set options.supportsSearchPath, but since options is "undefined" it "Cannot set properties of undefined".

Maybe your issue is similiar.
